File: frontend/mobile/lib/data/models/registration_request.dart
class RegistrationRequest {
  // User
  final String rsbsaId;
  final String firstName;
  final String lastName;
  final String password;
  final String? middleName;
  final String email;
  final String phoneNumber;

  // User Profile
  final String dateOfBirth; // format: dd-MM-yyyy
  final String gender;
  final String civilStatus;

  final String houseNo;
  final String street;
  final String barangay;
  final String municipality;
  final String province;
  final String region;

  final String farmerType;
  final double totalFarmAreaHa;

  RegistrationRequest({
    required this.rsbsaId,
    required this.firstName,
    required this.lastName,
    required this.password,
    this.middleName,
    required this.email,
    required this.phoneNumber,
    required this.dateOfBirth,
    required this.gender,
    required this.civilStatus,
    required this.houseNo,
    required this.street,
    required this.barangay,
    required this.municipality,
    required this.province,
    required this.region,
    required this.farmerType,
    required this.totalFarmAreaHa,
  });

  Map<String, dynamic> toJson() {
    return {
      'rsbsaId': rsbsaId,
      'firstName': firstName,
      'lastName': lastName,
      'password': password,
      'middleName': middleName,
      'email': email,
      'phoneNumber': phoneNumber,
      'dateOfBirth': dateOfBirth,
      'gender': gender,
      'civilStatus': civilStatus,
      'houseNo': houseNo,
      'street': street,
      'barangay': barangay,
      'municipality': municipality,
      'province': province,
      'region': region,
      'farmerType': farmerType,
      'totalFarmAreaHa': totalFarmAreaHa,
    };
  }
}

File: services/farmer/src/main/java/com/hashjosh/farmer/dto/RegistrationRequest.java
package com.hashjosh.farmer.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Set;
import java.util.UUID;

@Getter
@Setter
@Builder
public class RegistrationRequest {
    // User
    @NotNull(message = "Username cannot be null")
    @Size(max = 50, message = "Username cannot exceed 50 characters")
    private String rsbsaId;
    @NotNull(message = "Password cannot be null")
    @Size(min = 8, message = "Password must be at least 8 characters")
    private String password;
    @NotNull(message = "First name cannot be null")
    private String firstName;
    @NotNull(message = "Last name cannot be null")
    private String lastName;
    @NotNull(message = "Middle name cannot be null")
    private String middleName;
    @NotNull(message = "Email cannot be null")
    private String email;
    @NotNull(message = "Phone number cannot be null")
    private String phoneNumber;

    // User Profile
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateOfBirth;
    private String gender;
    private String civilStatus;

    private String street;
    private String barangay;
    private String municipality;
    private String province;
    private String region;

    private String farmerType;
    private Double totalFarmAreaHa;
}
